tools/workspace-plugin/src/conformance-rules/migration-groups/index.ts

```diff
@@ -0,0 +1,123 @@
+import { readJsonFile, workspaceRoot } from '@nx/devkit';
+import {
+  createConformanceRule,
+  type ProjectFilesViolation,
+} from '@nx/powerpack-conformance';
+import { existsSync } from 'node:fs';
+import { join } from 'node:path';
+import { satisfies } from 'semver';
+
+type Options = {
+  groups: Array<string[]>;
+  versionRange?: string;
+};
+
+export default createConformanceRule<Options>({
+  name: 'migration-groups',
+  category: 'consistency',
+  description:
+    'Ensures consistency across our project package.json files within the Nx repo',
+  reporter: 'project-files-reporter',
+  implementation: async ({ projectGraph, ruleOptions }) => {
+    const violations: ProjectFilesViolation[] = [];
+
+    for (const project of Object.values(projectGraph.nodes)) {
+      if (
+        project.name !== 'angular' &&
+        project.name !== 'eslint' &&
+        project.name !== 'storybook'
+      )
+        continue;
+      const migrationsPath = join(
+        workspaceRoot,
+        project.data.root,
+        'migrations.json'
+      );
+      if (existsSync(migrationsPath)) {
+        const migrations = readJsonFile(migrationsPath);
+        violations.push(
+          ...validateMigrations(
+            migrations,
+            project.name,
+            migrationsPath,
+            ruleOptions
+          )
+        );
+      }
+    }
+
+    return {
+      severity: 'high',
+      details: {
+        violations,
+      },
+    };
+  },
+});
+
+export function validateMigrations(
+  migrations: Record<string, unknown>,
+  sourceProject: string,
+  migrationsPath: string,
+  options: Options
+): ProjectFilesViolation[] {
+  if (!migrations.packageJsonUpdates) return [];
+
+  const violations: ProjectFilesViolation[] = [];
+
+  // Check that if package updates include one package in the group, then:
+  // 1. Every package from group is included
+  // 2. They all have the same version
+  for (const [key, value] of Object.entries(migrations.packageJsonUpdates)) {
+    if (!value.packages || !value.version) continue;
+    if (
+      options.versionRange &&
+      !satisfies(value.version, options.versionRange, {
+        includePrerelease: true,
+      })
+    )
+      continue;
+    const packages = Object.keys(value.packages);
+    for (const group of options.groups) {
+      if (!group.some((pkg) => packages.includes(pkg))) continue;
+      const result = group.reduce(
+        (acc, pkg) => {
+          if (packages.includes(pkg)) acc.present.push(pkg);
+          else acc.missing.push(pkg);
+          return acc;
+        },
+        { missing: [] as string[], present: [] as string[] }
+      );
+      if (result.missing.length > 0) {
+        violations.push({
+          message: `Package.json updates for "${key}" is missing packages in a group: ${result.missing.join(
+            ', '
+          )}. Versions of packages in a group must be in sync. Other packages in the group: ${result.present.join(
+            ', '
+          )}`,
```

Comment:
I'm not sure mentioning other packages in the group is relevant for this specific violation. Maybe the violation can be reworded so it lists the missing packages (as is doing right now) and the version that they should have (if there aren't more than 1).